File: TransCoder_tokenized_test_set_functions/java/SMALLEST_DERANGEMENT_SEQUENCE.java
static void generate_derangement ( int N ) {
  int S [ ] = new int [ N + 1 ];
  for ( int i = 1;
  i <= N;
  i ++ ) S [ i ] = i;
  int D [ ] = new int [ N + 1 ];
  for ( int i = 1;
  i <= N;
  i += 2 ) {
    if ( i == N ) {
      D [ N ] = S [ N - 1 ];
      D [ N - 1 ] = S [ N ];
    }
    else {
      D [ i ] = i + 1;
      D [ i + 1 ] = i;
    }
  }
  for ( int i = 1;
  i <= N;
  i ++ ) System . out . print ( D [ i ] + " " );
  System . out . println ( );
}

File: TransCoder_tokenized_test_set_functions/cpp/SMALLEST_DERANGEMENT_SEQUENCE.cpp
void generate_derangement ( int N ) {
  int S [ N + 1 ];
  for ( int i = 1;
  i <= N;
  i ++ ) S [ i ] = i;
  int D [ N + 1 ];
  for ( int i = 1;
  i <= N;
  i += 2 ) {
    if ( i == N && i % N != 0 ) {
      int temp = D [ N ] D [ N ] = D [ N - 1 ];
      D [ N - 1 ] = temp;
    }
    else {
      D [ i ] = i + 1;
      D [ i + 1 ] = i;
    }
  }
  for ( int i = 1;
  i <= N;
  i ++ ) printf ( "%d ", D [ i ] );
  printf ( "\n" );
}